### PR TITLE
Add detachDisk test to ITComputeTest.java

### PR DESCRIPTION
**Pull request by andreamlin**
Tests https://github.com/googleapis/google-cloud-java/issues/3735.

Blocked on 
- ~the fix in gax-java~
- ~a fix for another bug in gapic-generator to be released to artman~
- ~releasing gax-java~
- ~bumping the gax-java version in google-cloud-java~

### CODE REVIEW

**Comment by sduskis** [comment]
@andreamlin, it looks like there are a bunch of merge conflicts.

**Comment by andreamlin** [comment]
PTAL

fixed merge conflicts! test passes locally; hopefully the same can be said on kokoro

File: google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java [hunk 1]
```diff
@@ -45,31 +70,200 @@
 
 public class ITComputeTest {
 
-  private static final String ZONE = "us-central1-a";
+  private static final String ZONE = "us-west1-a";
```
Comment (sduskis) [inline google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java]: Would it make sense to move REGION near here?

File: google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java [hunk 2]
```diff
@@ -45,31 +70,200 @@
...
-    diskTypeSettings =
+    DiskTypeSettings diskTypeSettings =
         DiskTypeSettings.newBuilder()
             .setCredentialsProvider(FixedCredentialsProvider.create(credentials))
             .build();
     diskTypeClient = DiskTypeClient.create(diskTypeSettings);
+
+    InstanceSettings instanceSettings =
+        InstanceSettings.newBuilder()
+            .setCredentialsProvider(FixedCredentialsProvider.create(credentials))
```
Comment (sduskis) [inline google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java]: Can we reuse the `FixedCredentialsProvider` for all three calls?

File: google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java [hunk 3]
```diff
@@ -45,31 +70,200 @@
...
+      subNetworkClient.close();
+      diskClient.close();
+      imageClient.close();
+    }
+  }
+
+  @Test
+  public void testDetachDisk() throws Exception {
+    long startAt = System.currentTimeMillis();
+    System.out.println("Clients created in " + (System.currentTimeMillis() - startAt) + "ms.");
```
Comment (sduskis) [inline google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java]: Does this print statement still belong here?

File: google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java [hunk 4]
```diff
@@ -45,31 +70,200 @@
...
+    // inserted network).
+    Instance instanceResource =
+        Instance.newBuilder()
+            .setName(INSTANCE_NAME.getInstance())
+            .addNetworkInterfaces(networkInterface)
+            .setMachineType(MACHINE_TYPE_NAME.toString())
+            .addDisks(bootDisk)
+            .build();
+    Operation op1 = instanceClient.insertInstance(PROJECT_ZONE_NAME, instanceResource);
+    System.out.println(String.format("Instance created: %s", op1.toString()));
```
Comment (sduskis) [inline google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java]: Does this still belong here?  If so, then perhaps a log statement would be better.

File: google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java [hunk 5]
```diff
@@ -45,31 +70,200 @@
...
+
+    // Insert a disk.
+    Disk disk =
+        Disk.newBuilder()
+            .setName(DISK_NAME.getDisk())
+            .setSizeGb(String.valueOf(7))
+            .setType(diskTypeName.toString())
+            .build();
+    Operation op2 = diskClient.insertDisk(PROJECT_ZONE_NAME, disk);
+    System.out.println(String.format("Disk created: %s", op2.toString()));
```
Comment (sduskis) [inline google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java]: Log, and sleep?

File: google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java [hunk 6]
```diff
@@ -45,31 +70,200 @@
...
+        AttachedDisk.newBuilder()
+            .setDeviceName(DISK_NAME.getDisk())
+            .setAutoDelete(false)
+            .setMode("READ_WRITE")
+            .setSource(DISK_NAME.toString())
+            .build();
+    Operation op3 = instanceClient.attachDiskInstance(INSTANCE_NAME, false, attachedDisk);
+    System.out.println(String.format("Disk attached: %s", op3.toString()));
+    Thread.sleep(10000);
+    System.out.println("----------------------------------------");
```
Comment (sduskis) [inline google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java]: log statement?

File: google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java [hunk 7]
```diff
@@ -45,31 +70,200 @@
...
+
+    Operation op4 = instanceClient.detachDiskInstance(INSTANCE_NAME, DISK_NAME.getDisk());
+    System.out.println(String.format("Disk detached: %s", op4.toString()));
```
Comment (sduskis) [inline google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java]: log statement.

**Comment by andreamlin** [comment]
Blocked on https://github.com/googleapis/google-cloud-java/issues/4501 (LRO for Compute)

**Comment by sduskis** [comment]
@andreamlin, it looks like this PR has gone stale.  Please reopen once the blockers are removed.

**Pull request closed without merging**